x-pack/test/api_integration/apis/ml/data_frame_analytics/delete.ts

```diff
@@ -0,0 +1,237 @@
+    });
+
+    describe('DeleteDataFrameAnalytics', () => {
+      it('should delete analytics jobs by id', async () => {
+        const { body } = await supertest
+          .delete(`/api/ml/data_frame/analytics/${jobId}_1`)
+          .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+          .set(COMMON_HEADERS)
+          .expect(200);
+        expect(body.analyticsJobDeleted.success).to.eql(true);
+      });
+
+      it('should not allow to retrieve analytics jobs for the user without required permissions', async () => {
+        const { body } = await supertest
+          .delete(`/api/ml/data_frame/analytics/${jobId}_2`)
+          .auth(USER.ML_UNAUTHORIZED, ml.securityCommon.getPasswordForUser(USER.ML_UNAUTHORIZED))
+          .set(COMMON_HEADERS)
+          .expect(404);
+
+        expect(body.error).to.eql('Not Found');
+        expect(body.message).to.eql('Not Found');
+
+        const { body: body2 } = await supertest
+          .delete(`/api/ml/data_frame/analytics/${jobId}_2`)
+          .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+          .set(COMMON_HEADERS)
+          .expect(404);
+
+        expect(body2.error).to.eql('Not Found');
+        expect(body2.message).to.eql('Not Found');
+      });
+
+      it('should show 404 error if job does not exist or has already been deleted', async () => {
+        const { body } = await supertest
+          .delete(`/api/ml/data_frame/analytics/${jobId}_invalid`)
+          .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+          .set(COMMON_HEADERS)
+          .expect(404);
+
+        expect(body.error).to.eql('Not Found');
+        expect(body.message).to.eql('Not Found');
+      });
+
+      describe('with deleteTargetIndex setting', function () {
+        const analyticsId = `${jobId}_2`;
+        const destinationIndex = generateDestinationIndex(analyticsId);
+
+        before(async () => {
+          await es.indices.create({ index: destinationIndex });
+          await ml.api.assertIndicesExist(generateDestinationIndex(analyticsId));
+        });
+
+        after(async () => {
+          await es.indices.delete({ index: destinationIndex, ignore: [404] });
+        });
+
+        it('should delete job and target index by id', async () => {
+          const { body } = await supertest
+            .delete(`/api/ml/data_frame/analytics/${analyticsId}`)
+            .query({ deleteTargetIndex: true })
+            .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+            .set(COMMON_HEADERS)
+            .expect(200);
+
+          expect(body.analyticsJobDeleted.success).to.eql(true);
+          expect(body.targetIndexDeleted.success).to.eql(true);
+          expect(body.targetIndexPatternDeleted.success).to.eql(false);
+          expect(jobDeleted(analyticsId)).to.eql(true);
+
+          const targetIndexExists = await es.indices.exists({
```

Comment:
I'd rather have this as a service method in `ml.api`. Something like `waitForIndexNotToExist` with the code that's already part of [`deleteIndices`](https://github.com/elastic/kibana/blob/master/x-pack/test/functional/services/ml/api.ts#L127)